### PR TITLE
fix: Fix issues found during replaying events and dumping state

### LINKED ISSUE #12334: Validate and Fix state dumpers for `MONO_POST_EVENT_STREAM_REPLAY` when comparing with `MOD_POST_EVENT_STREAM_REPLAY`

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #12334 

- Fixed NPEs in dumpers
- Changed to not use NftID.DEFAULT in any migration code as key
- Changed all stores in. token service to add a check when put into state to have key that is not DEFAULT
- Added StateDumper to dump all three schedule virtual maps. Actual Dumping and validating scheduled state will be done in next PR
- Fixed token relation linkedLists in `NFTOwnerChangesStep`

### CODE REVIEW

**Review by iwsimon — APPROVED** [review]
LGTM. Thanks @Neeharika-Sompalli

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/AccountStateTranslator.java [hunk 1]
```diff
@@ -173,12 +177,16 @@ public static Account accountFromOnDiskAccount(@NonNull final OnDiskAccount acco
                 .stakedToMe(account.getStakedToMe())
                 .stakePeriodStart(account.getStakePeriodStart())
                 .firstContractStorageKey(firstContractStorageKey)
-                .headNftId(NftID.newBuilder()
-                        .tokenId(TokenID.newBuilder()
-                                .tokenNum(account.getHeadNftTokenNum())
-                                .realmNum(StaticProperties.getRealm())
-                                .shardNum(StaticProperties.getShard()))
-                        .serialNumber(account.getHeadNftSerialNum()))
+                .headNftId(
```
Comment (tinker-michaelj) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/AccountStateTranslator.java:180]: Great catch. 🙌

Created an [issue](https://github.com/hashgraph/hedera-services/issues/12548) for us to review all migrations one more time for this kind of problem.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java [hunk 2]
```diff
@@ -122,12 +123,25 @@ protected Map<EntityIDPair, Long> tokenRelChangesFrom(
 
             // If the token rel's balance has changed, add it to the list of changes
             final var netFungibleChange = modifiedTokenRelBalance - persistedBalance;
-            if (netFungibleChange != 0 || !filterZeroAmounts) {
-                tokenChanges.put(modifiedRel, netFungibleChange);
+            if (netFungibleChange != 0) {
+                tokenRelChanges.put(modifiedRel, netFungibleChange);
+            } else {
+                // It is possible that we have updated pointers in the token relation during crypto transfer
```
Comment (tinker-michaelj) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java:129]: Nice comment!

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java [hunk 3]
```diff
@@ -83,6 +83,7 @@ protected WritableKVState<ProtoBytes, AccountID> aliases() {
      */
     public void put(@NonNull final Account account) {
         Objects.requireNonNull(account);
+        requireNotDefault(account.accountIdOrThrow());
```
Comment (tinker-michaelj) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java:86]: Did this fix a bug, or just a sensible defensive check?

**Review by tinker-michaelj — DISMISSED** [review]
LGTM, tyvm @Neeharika-Sompalli !

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java (hunk 3 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java:86]: There was a case where we set `NftID.DEFAULT` set in migration code previously. Just a sensible check. Will do in all stores in later PR  😃

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/AccountStateTranslator.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/AccountStateTranslator.java:180]: Makes sense. tyvm 🙌

**Review by iwsimon — APPROVED** [review]
LGTM. Thanks @Neeharika-Sompalli

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java (hunk 3 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java:86]: Ah got it. ☺️

**Pull request merged**